pbj-core/pbj-runtime/src/test/java/com/hedera/pbj/runtime/io/ReadableSequentialTestBase.java

```diff
@@ -0,0 +1,207 @@
+package com.hedera.pbj.runtime.io;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.nio.charset.StandardCharsets;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+public abstract class ReadableSequentialTestBase extends ReadableTestBase {
+
+    @NonNull
+    @Override
+    protected abstract ReadableSequentialData emptySequence();
+
+    @NonNull
+    @Override
+    protected abstract ReadableSequentialData fullyUsedSequence();
+
+    @Override
+    @NonNull
+    protected abstract ReadableSequentialData sequence(@NonNull byte [] arr);
+
+    @Test
+    @DisplayName("Stream with no data")
+    void noDataTest() throws Exception {
```

Comment:
Even in test methods, declaring `throws Exception` is not great, as it can hide unexpected changes in API.  Generally better to declare `throws` only as necessary.

Reply:
This "throws Exception" clause comes from `AutoCloseable` interface. Originally these tests were enabled only for `ReadableStreamingData`, which is known to implement `AutoCloseable`, so the tests just used try-with-resources. Now I'm trying to make the tests more generic, so they can be used for any `ReadableSequentialData`, e.g. `BufferedData`, but not all `ReadableSequentialData` are `AutoCloseable`.

The only (slightly ugly) option I see is to drop "throws Exception" from all test methods, but instead explicitly catch exceptions from `AutoCloseable.close()` and re-throw them as runtime exceptions. What do you think about it?

Reply:
I would actually recommend using the `Closeable` interface in the test here (which anything dealing with IO, such as PBJ code, should implement, rather than the parent interface AutoCloseable). That only throws IOException, which is pretty close to what we actually might throw from the code.

If the PBJ classes don't implement `Closeable`, we should change them to do so (`Closeable` extends `AutoCloseable`).

Reply:
Do I read it correctly, is your proposal to not close anything in the tests (aside from Closeable vs AutoCloseable, which should be taken care of in a separate fix)?

Reply:
I propose to _add an issue_ to replace `AutoCloseable` with `Closeable` as the interface implemented in PBJ I/O classes.  As part of that issue we can replace `Exception` with `IOException` on the tests.
We should not change the behavior of the tests.

Reply:
I filed https://github.com/hashgraph/pbj/issues/134 about it. Note that even after that issue is fixed, we won't be able to use try-with-resources in all tests, only some of them. The problem is `Closeable` (or `AutoCloseable` today) is at the class level, `ReadableStreamingData` and `WriteableStreamingData`, while tests work with interfaces `ReadableSequentialData` and `WriteableSequentialData`, which are not closeable. That's why we will still have extra instanceof checks.